worldwind/src/main/java/gov/nasa/worldwind/render/RenderContext.java

```diff
@@ -559,12 +559,15 @@ public boolean hasUserProperty(Object key) {
 
         public TextCacheKey set(String text, TextAttributes attributes) {
             this.text = text;
-            this.textColor = (attributes != null ? attributes.getTextColor() : null);
-            this.textSize = (attributes != null ? attributes.getTextSize() : 0);
-            this.typeface = (attributes != null ? attributes.getTypeface() : null);
-            this.enableOutline = (attributes != null && attributes.isEnableOutline());
-            this.outlineColor = (attributes != null ? attributes.getOutlineColor() : null);
-            this.outlineWidth = (attributes != null ? attributes.getOutlineWidth() : 0);
+            if (attributes != null) {
```

Comment:
Cache key properties need appropriate values when the attributes bundle is null. Consider the following state transitions:
• null -> non-null
• null -> null
• non-null -> null
• non-null -> non-null
Most properties can be assigned to either null or the appropriate value. Since we want to avoid repeated allocations, Color properties need to be assigned to null, assigned to a new instance and set, or just set, depending on the state transition.

Reply:
You're right! I forgot about the scratch key utilization! I'll get it sorted out to account for the reuse case.

Reply:
I like your implementation. The branching is simple and reads very well.